ls1intum/Artemis
# [Bugfix] Edit exam quiz when test run exists

### Checklist

### Motivation and Context


See [#2961](https://github.com/ls1intum/Artemis/issues/2961)

### Description
Whenever a test run exists for an exam (before the exam has started), and there are quiz exercise submissions for the given test exam, the instructor can't edit the answer options in the markdown. In the implemented change the instructor will see warnings next to the edit button in the exercise group dashboard and the quiz exercise edit page.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create an exam with an upcoming start date and an exercise group with a quiz exercise.
4. Create a test run for the exam with the said quiz exercise.
5. Navigate to the exercise dashboard and check if the warning is present next to the "Edit" button
6. Click on Edit and check if the same warning is present and the save button is disabled.
7. Test accordingly if the warnings aren't there when the test run gets deleted.

### Test Coverage



ExamService.java: 94%
ExerciseService.java: 89%

### Screenshots


![image](https://user-images.githubusercontent.com/63168603/112764945-76fcf600-9013-11eb-93f5-fb5227662c23.png)
![image](https://user-images.githubusercontent.com/63168603/113023127-00383800-918e-11eb-97e4-ae6bbade21f9.png)
